### PR TITLE
allows for the exec_service pod image to be pulled from a custom repo

### PR DESCRIPTION
**Pull request by rackeric**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Some clusters may not be able to reach the upstream Kubernetes image repos, like registry.k8s.io, but can reach some internal mirror of these. This will allow to set a custom image repo for the exec_service that cluster loader uses to create the exec pods. You can supply the image repo via a clusterloader2 CLI flag `--registry-k8s-repo`

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/perf-tests/issues/2704

### CODE REVIEW

File: clusterloader2/pkg/config/cluster.go [hunk 1]
```diff
@@ -61,7 +61,8 @@ type ClusterConfig struct {
 // ExecServiceConfig represents all flags used by service config.
 type ExecServiceConfig struct {
 	// Determines if service config should be enabled.
-	Enable bool
+	Enable 			bool
+	RegistryK8sRepo string
```
Comment (vincepri) [inline clusterloader2/pkg/config/cluster.go:65]: Could we run gofmt on these?

File: clusterloader2/pkg/execservice/exec_service.go [hunk 2]
```diff
@@ -67,6 +67,12 @@ func InitFlags(c *config.ExecServiceConfig) {
 		true,
 		"Whether to enable exec service that allows executing arbitrary commands from a pod running in the cluster.",
 	)
+	flags.StringVar(
+		&c.RegistryK8sRepo,
```
Comment (vincepri) [inline clusterloader2/pkg/execservice/exec_service.go:71]: Should we just call this `ImageRegistry` instead?

File: clusterloader2/pkg/config/cluster.go (hunk 1 above)
Comment (vincepri) [inline clusterloader2/pkg/config/cluster.go:65]:
```suggestion
	ImageRegistry string
```

**Comment by vincepri** [comment]
@wojtek-t @mboersma would you be able to take a look?

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by rackeric** [comment]
@wojtek-t @mboersma I rebased my changes and lost the LGTM tags. but also, these tests seem to be failing for things unrelated to my change. how could I troubleshoot this? CC @vincepri

File: clusterloader2/pkg/execservice/exec_service.go [hunk 3]
```diff
@@ -82,6 +88,7 @@ func SetUpExecService(f *framework.Framework, _ config.ExecServiceConfig) error
 	mapping["Name"] = execDeploymentName
 	mapping["Namespace"] = execDeploymentNamespace
 	mapping["Replicas"] = execPodReplicas
+	mapping["ImageRegistry"] = c.ImageRegistry
```
Comment (wojtek-t) [inline clusterloader2/pkg/execservice/exec_service.go:91]: Test failures are relevant - there is no `c` variable here....

Comment (rackeric) [reply clusterloader2/pkg/execservice/exec_service.go:91]: ack, thanks. getting it fixed now.

Comment (rackeric) [reply clusterloader2/pkg/execservice/exec_service.go:91]: all tests are passing now.

**Pull request merged**